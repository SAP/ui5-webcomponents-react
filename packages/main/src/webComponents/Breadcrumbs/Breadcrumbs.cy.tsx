import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { BreadcrumbsItem } from '../BreadcrumbsItem';
import { Breadcrumbs } from './index';

describe('Breadcrumbs', () => {
  it('Basic Test (generated)', () => {
    cy.mount(
      <Breadcrumbs>
        <BreadcrumbsItem>Hello World!</BreadcrumbsItem>
      </Breadcrumbs>
    );
    cy.get('[ui5-breadcrumbs]').should('exist');
  });

  cypressPassThroughTestsFactory(Breadcrumbs, { children: <BreadcrumbsItem>Hello World!</BreadcrumbsItem> });
});
