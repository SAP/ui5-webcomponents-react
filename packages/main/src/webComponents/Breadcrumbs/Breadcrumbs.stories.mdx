import { Canvas, Meta, Story } from '@storybook/addon-docs';
import { ArgsTableWithNote } from '@docs/ArgsTableWithNote';
import { Breadcrumbs } from './index';
import { BreadcrumbsDesign } from '../../enums/BreadcrumbsDesign';
import { BreadcrumbsSeparatorStyle } from '../../enums/BreadcrumbsSeparatorStyle';
import { BreadcrumbsItem } from '../BreadcrumbsItem';
import { DocsHeader } from '@docs/DocsHeader';
import { Footer } from '@docs/Footer';
import Description from './BreadcrumbsDescription.md';
import SubcomponentsSection from '../../../../../.storybook/docs/SubcomponentsSection.md';
import BreadcrumbsItemDescription from '../BreadcrumbsItem/BreadcrumbsItemDescription.md';

<Meta
  title="Layouts & Floorplans / Breadcrumbs"
  component={Breadcrumbs}
  subcomponents={{ BreadcrumbsItem }}
  argTypes={{
    children: { control: { disable: true } }
  }}
  args={{
    design: BreadcrumbsDesign.Standard,
    separatorStyle: BreadcrumbsSeparatorStyle.Slash
  }}
/>

<DocsHeader since="0.18.0" />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return (
        <Breadcrumbs {...args}>
          <BreadcrumbsItem>Products</BreadcrumbsItem>
          <BreadcrumbsItem>Hardware</BreadcrumbsItem>
          <BreadcrumbsItem>Notebooks</BreadcrumbsItem>
        </Breadcrumbs>
      );
    }}
  </Story>
</Canvas>

## Properties

<ArgsTableWithNote story="." />

<Description />

<SubcomponentsSection />

## BreadcrumbsItem

<BreadcrumbsItemDescription />

<Footer />
