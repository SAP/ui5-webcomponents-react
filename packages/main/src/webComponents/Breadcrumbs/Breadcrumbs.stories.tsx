import type { Meta, StoryObj } from '@storybook/react';
import { BreadcrumbsDesign, BreadcrumbsSeparatorStyle } from '../../enums/index.js';
import { BreadcrumbsItem } from '../BreadcrumbsItem/index.js';
import { Breadcrumbs } from './index.js';

const meta = {
  title: 'Layouts & Floorplans / Breadcrumbs',
  component: Breadcrumbs,
  argTypes: {
    children: { control: { disable: true } }
  },
  args: {
    design: BreadcrumbsDesign.Standard,
    separatorStyle: BreadcrumbsSeparatorStyle.Slash
  }
} satisfies Meta<typeof Breadcrumbs>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: (args) => {
    return (
      <Breadcrumbs {...args}>
        <BreadcrumbsItem>Products</BreadcrumbsItem>
        <BreadcrumbsItem>Hardware</BreadcrumbsItem>
        <BreadcrumbsItem>Notebooks</BreadcrumbsItem>
      </Breadcrumbs>
    );
  }
};
