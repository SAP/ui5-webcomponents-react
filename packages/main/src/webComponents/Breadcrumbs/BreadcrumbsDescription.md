## Keyboard Handling

The `Breadcrumbs` provides advanced keyboard handling.  

*   \[F4, ALT+UP, ALT+DOWN, SPACE, ENTER\] - If the dropdown arrow is focused - opens/closes the drop-down.
*   \[SPACE, ENTER\] - Activates the focused item and triggers the `onItemClick` event.
*   \[ESC\] - Closes the drop-down.
*   \[LEFT\] - If the drop-down is closed - navigates one item to the left.
*   \[RIGHT\] - If the drop-down is closed - navigates one item to the right.
*   \[UP\] - If the drop-down is open - moves focus to the next item.
*   \[DOWN\] - If the drop-down is open - moves focus to the previous item.
*   \[HOME\] - Navigates to the first item.
*   \[END\] - Navigates to the last item.