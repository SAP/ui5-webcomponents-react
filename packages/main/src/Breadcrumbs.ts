import { deprecationNotice } from '@ui5/webcomponents-react-base';
import type { BreadcrumbsDomRef, BreadcrumbsPropTypes } from './webComponents/Breadcrumbs';
import { Breadcrumbs } from './webComponents/Breadcrumbs';

deprecationNotice(
  'Breadcrumbs',
  `importing from '@ui5/webcomponents-react/dist/Breadcrumbs' is deprecated. Please use "import { Breadcrumbs } from '@ui5/webcomponents-react';" instead.`
);

export { Breadcrumbs };

export type { BreadcrumbsPropTypes, BreadcrumbsDomRef };
