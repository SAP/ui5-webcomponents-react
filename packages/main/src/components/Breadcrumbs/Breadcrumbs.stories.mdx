import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import { Breadcrumbs } from '@ui5/webcomponents-react/dist/Breadcrumbs';
import { BreadcrumbsSeparatorStyle } from '@ui5/webcomponents-react/dist/BreadcrumbsSeparatorStyle';
import { Link } from '@ui5/webcomponents-react/dist/Link';
import { DocsHeader } from '@shared/stories/DocsHeader';

<Meta
  title="Layouts & Floorplans / Breadcrumbs"
  component={Breadcrumbs}
  argTypes={{
    children: { control: { disable: true } }
  }}
  args={{
    separatorStyle: BreadcrumbsSeparatorStyle.Slash,
    currentLocationText: 'Super Slim Notebook 13 inch',
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return (
        <Breadcrumbs {...args}>
          <Link>Products</Link>
          <Link>Hardware</Link>
          <Link>Notebooks</Link>
          <Link>13 inch</Link>
        </Breadcrumbs>
      );
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
