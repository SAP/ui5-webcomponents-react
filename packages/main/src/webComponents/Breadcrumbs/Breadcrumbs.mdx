import { ArgTypesWithNote, ControlsWithNote, DocsHeader, Footer } from '@sb/components';
import SubcomponentsSection from '@sb/docs/SubcomponentsSection.md?raw';
import { Canvas, Markdown, Meta } from '@storybook/blocks';
import { BreadcrumbsItem } from '../BreadcrumbsItem';
import BreadcrumbsItemDescription from '../BreadcrumbsItem/BreadcrumbsItemDescription.md?raw';
import * as ComponentStories from './Breadcrumbs.stories';
import Description from './BreadcrumbsDescription.md?raw';
import { excludePropsForAbstract } from '@sb/utils';

<Meta of={ComponentStories} />

<DocsHeader since="0.18.0" />

<br />

## Example

<Canvas of={ComponentStories.Default} />

## Properties

<ControlsWithNote of={ComponentStories.Default} />

<Markdown>{Description}</Markdown>

<Markdown>{SubcomponentsSection}</Markdown>

## BreadcrumbsItem

<Markdown>{BreadcrumbsItemDescription}</Markdown>
<ArgTypesWithNote exclude={excludePropsForAbstract} hideHTMLPropsNote of={BreadcrumbsItem} />

<Footer />
