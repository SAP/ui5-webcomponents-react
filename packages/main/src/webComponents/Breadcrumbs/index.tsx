'use client';

import '@ui5/webcomponents/dist/Breadcrumbs.js';
import { ReactNode } from 'react';
import { BreadcrumbsDesign, BreadcrumbsSeparatorStyle } from '../../enums';
import { CommonProps } from '../../interfaces/CommonProps';
import { Ui5CustomEvent } from '../../interfaces/Ui5CustomEvent';
import { Ui5DomRef } from '../../interfaces/Ui5DomRef';
import { withWebComponent } from '../../internal/withWebComponent';

interface BreadcrumbsAttributes {
  /**
   * Defines the visual indication and behavior of the breadcrumbs. Available options are `Standard` (by default) and `NoCurrentPage`.
   *
   * **Note:** The `Standard` breadcrumbs show the current page as the last item in the trail. The last item contains only plain text and is not a link.
   */
  design?: BreadcrumbsDesign | keyof typeof BreadcrumbsDesign;
  /**
   * Determines the visual style of the separator between the breadcrumb items.
   *
   * Available options are:
   *
   * *   `Slash`
   * *   `BackSlash`
   * *   `DoubleBackSlash`
   * *   `DoubleGreaterThan`
   * *   `DoubleSlash`
   * *   `GreaterThan`
   */
  separatorStyle?: BreadcrumbsSeparatorStyle | keyof typeof BreadcrumbsSeparatorStyle;
}

export interface BreadcrumbsDomRef extends BreadcrumbsAttributes, Ui5DomRef {}

export interface BreadcrumbsPropTypes extends BreadcrumbsAttributes, CommonProps {
  /**
   * Defines the component items.
   *
   * **Note:** Use the `BreadcrumbsItem` component to define the desired items.
   */
  children?: ReactNode | ReactNode[];
  /**
   * Fires when a `BreadcrumbsItem` is clicked. **Note:** You can prevent browser location change by calling `event.preventDefault()`.
   */
  onItemClick?: (
    event: Ui5CustomEvent<
      BreadcrumbsDomRef,
      { item: HTMLElement; altKey: boolean; ctrlKey: boolean; metaKey: boolean; shiftKey: boolean }
    >
  ) => void;
}

/**
 * Enables users to navigate between items by providing a list of links to previous steps in the user's navigation path. It helps the user to be aware of their location within the application and allows faster navigation.
 *
 * The last three steps can be accessed as links directly, while the remaining links prior to them are available in a drop-down menu.
 *
 * You can choose the type of separator to be used from a number of predefined options.
 *
 * __Note:__ This component is a web component developed by the UI5 Web Componentsâ€™ team.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/Breadcrumbs" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const Breadcrumbs = withWebComponent<BreadcrumbsPropTypes, BreadcrumbsDomRef>(
  'ui5-breadcrumbs',
  ['design', 'separatorStyle'],
  [],
  [],
  ['item-click'],
  () => import('@ui5/webcomponents/dist/Breadcrumbs.js')
);

Breadcrumbs.displayName = 'Breadcrumbs';

Breadcrumbs.defaultProps = {
  design: BreadcrumbsDesign.Standard,
  separatorStyle: BreadcrumbsSeparatorStyle.Slash
};

export { Breadcrumbs };
