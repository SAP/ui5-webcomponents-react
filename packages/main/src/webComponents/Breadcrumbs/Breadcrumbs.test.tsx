import { render } from '@shared/tests';
import { Breadcrumbs } from '@ui5/webcomponents-react/dist/Breadcrumbs';
import React from 'react';

describe('Breadcrumbs', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(<Breadcrumbs />);
    expect(asFragment()).toMatchSnapshot();
  });
});
