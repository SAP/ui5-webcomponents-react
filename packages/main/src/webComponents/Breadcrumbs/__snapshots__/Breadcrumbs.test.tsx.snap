// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-breadcrumbs
    design="Standard"
    separator-style="Slash"
    ui5-breadcrumbs=""
  />
</DocumentFragment>
`;
